src/object_properties.cpp

```diff
@@ -167,4 +169,5 @@ void ObjectProperties::deSerialize(std::istream &is)
 	eye_height = readF32(is);
 	zoom_fov = readF32(is);
 	use_texture_alpha = readU8(is);
+	show_selection_box = readU8(is);
```

Comment:
This will read `0` (or `false`) on older servers. Check the remaining data bytes or maybe a try/catch is needed here for compatibility.

Reply:
Which server versions can have this problem?
I tried to fix it, tell me if it's ok.

Reply:
Basically any that were build prior this PR. The client expects a byte there, but that isn't sent by the server. Thus, it will either default to 0 or throw some exception.
To test this use an older server build and join with a new client. The client should see the selection boxes.